File: app/src/main/java/dev/ragnarok/fenrir/api/model/VKApiCall.java
package dev.ragnarok.fenrir.api.model;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

public class VKApiCall implements VKApiAttachment {

    @SerializedName("initiator_id")
    public int initiator_id;

    @SerializedName("receiver_id")
    public int receiver_id;

    @SerializedName("state")
    public String state;

    @SerializedName("time")
    public long time;

    @NonNull
    @Override
    public String getType() {
        return TYPE_CALL;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/VKApiCall.kt
package dev.ragnarok.fenrir.api.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class VKApiCall : VKApiAttachment {
    @SerialName("initiator_id")
    var initiator_id = 0

    @SerialName("receiver_id")
    var receiver_id = 0

    @SerialName("state")
    var state: String? = null

    @SerialName("time")
    var time: Long = 0
    override fun getType(): String {
        return VKApiAttachment.TYPE_CALL
    }
}